MMM-MotionDetector.js

```diff
@@ -68,35 +73,54 @@ Module.register("MMM-MotionDetector", {
       video: video,
       captureIntervalTime: this.config.captureIntervalTime,
       motionCanvas: canvas,
+
       initSuccessCallback: () => {
         Log.info("MMM-MotionDetector: DiffCamEngine init successful.");
+        if (this.config.debug) {
+					Log.info("Timeout set to " + this.config.timeout + " milliseconds.");
+					Log.info("Motion threshold set to " + this.config.scoreThreshold);
+					Log.info("Will MotionDetector control the monitor? " + this.config.controlDisplay);
+					if (this.config.additionalNotification) { Log.info("Will send additional notification: " + this.config.additionalNotification); }
+				}
         DiffCamEngine.start();
       },
       initErrorCallback: (error) => {
         Log.error("MMM-MotionDetector: DiffCamEngine init failed: " + error);
         this.error = error;
         this.updateDom();
       },
+
       captureCallback: (payload) => {
         const score = payload.score;
         const currentDate = new Date();
-        this.percentagePoweredOff = ((100 * this.poweredOffTime) / (currentDate.getTime() - this.timeStarted)).toFixed(
-          2
-        );
-        if (score > this.config.scoreThreshold) {
+        this.percentagePoweredOff = ((100 * this.poweredOffTime) / (currentDate.getTime() - this.timeStarted)).toFixed(2);
+        const time = currentDate.getTime() - this.lastTimeMotionDetected.getTime();
+
+        if (score > this.config.scoreThreshold) { // We have found motion
           Log.info("MMM-MotionDetector: Motion detected, score " + score);
           this.sendSocketNotification("MOTION_DETECTED", { score: score });
-          this.sendNotification("MOTION_DETECTED", { score: score });
+
           if (this.poweredOff) {
-            this.sendSocketNotification("ACTIVATE_MONITOR");
             this.poweredOffTime = this.poweredOffTime + (currentDate.getTime() - this.lastTimePoweredOff.getTime());
+            this.sendNotification("MOTION_DETECTED", { score: score });
```

Comment:
why send this notification only when the monitor is poweredOff? SHouldnt it be send always when a motion is detected?

Reply:
I mentally chewed on this for quite a while. The idea is that it only sends the notification after the timeout value. Otherwise you're just broadcasting the notification all the time and the timeout threshold has no effect. I suppose it really depends on how you're expecting to use the module to signal other modules.

Reply:
I thought you wanted to have the notification send always when a motion is detected. The timeout can then be handled by the listener of the notification